shelf/bucket_update/search_updater.py

```diff
@@ -57,9 +58,46 @@ def run(self):
             self.update_manager.bulk_update(bulk_update)
             all_id_list = all_id_list + bulk_update.keys()
 
+        self.remove_unlisted_documents_per_bucket(all_id_list)
+
+    def prune(self):
+        """
+            Loops through a list of artifact paths, compares them against
+            the search index, and removes any indexes that no longer exist.
+        """
+        path_list = []
+
+        try:
+            path_list.extend(self.load_path_list())
+        except BucketNotFoundError:
+            self.logger.info("Bucket wasn't found. Leaving path_list empty.")
+
+        gc.collect()
+        self.logger.info("Starting to collect {0} artifact's Elasticsearch ids.".format(len(path_list)))
+        all_id_list = []
+
+        for chunk_list in self._chunk(path_list):
+            for path in chunk_list:
+                identity = self.container.resource_identity_factory \
+                    .from_cloud_identifier(path)
+                all_id_list.append(identity.search)
+
+        self.remove_unlisted_documents_per_bucket(all_id_list)
+
+    def remove_unlisted_documents_per_bucket(self, all_id_list):
+        """
+            This is essentially syntactic sugar for calling
+            update_manager.remove_unlisted_documents_per_bucket() with some
+            log statements.
+
+            Args:
+                all_id_list(List(basestring)): List containing all the
```

Comment:
I looked for an example elsewhere in Shelf. I can't remember exactly where, but it definitely is elsewhere.